src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -267,6 +267,28 @@ fn check_contract_call(checker: &mut TypeChecker, args: &[SymbolicExpression], c
     Ok(expected_sig.returns)
 }
 
+fn check_trait_principal(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(1, args)?;
+
+    if let SymbolicExpressionType::Atom(trait_instance) = &args[0].expr {
```

Comment:
:nail_care: we usually like using guard condition patterns, such as:
```
let trait_instance = match &args[0].expr {
  SymbolicExpressionType::Atom(trait_instance) => trait_instance,
  _ => return Err(CheckError::new(CheckErrors::TraitPrincipalExpectsTrait))
};

...
```

Reply:
sure